lib/project_types/rails/gem.rb

```diff
@@ -15,42 +15,94 @@ def install(ctx, *args)
         version = args.shift
         gem = new(ctx: ctx, name: name, version: version)
         ctx.debug(ctx.message('rails.gem.installed_debug', name, gem.installed?))
-        gem.install! unless gem.installed?
+        gem.installed? ? true : gem.install!
       end
 
       def binary_path_for(ctx, binary)
-        File.join(gem_home(ctx), 'bin', binary)
+        path_to_binary = File.join(gem_home(ctx), 'bin', binary)
+        File.exist?(path_to_binary) ? path_to_binary : binary
       end
 
       def gem_home(ctx)
         ctx.getenv('GEM_HOME') || apply_gem_home(ctx)
       end
 
+      def gem_path(ctx)
+        ctx.getenv('GEM_PATH') || apply_gem_path(ctx)
+      end
+
       private
 
       def apply_gem_home(ctx)
         path = File.join(ctx.getenv('HOME'), '.gem', 'ruby', RUBY_VERSION)
         ctx.mkdir_p(path) unless Dir.exist?(path)
+        ctx.debug(ctx.message('rails.gem.setting_gem_home', path))
         ctx.setenv('GEM_HOME', path)
       end
+
+      def apply_gem_path(ctx)
+        path = ''
+        out, stat = ctx.capture2('gem', 'environment', 'path')
+        if stat
+          path = out&.empty? ? '' : out.strip
+        end
+        ctx.debug(ctx.message('rails.gem.setting_gem_path', path))
+        ctx.setenv('GEM_PATH', path)
+      end
     end
 
     def installed?
-      !!Dir.glob("#{self.class.gem_home(ctx)}/gems/#{name}-*").detect do |f|
-        f =~ %r{/#{Regexp.quote(name)}-\d}
+      ctx.debug(
+        ctx.message('rails.gem.checking_installation_path', "#{self.class.gem_home(ctx)}/gems/", name)
+      )
+      found = !!Dir.glob("#{self.class.gem_home(ctx)}/gems/#{name}-*").detect do |f|
+        gem_satisfies_version?(f)
       end
+      unless found
+        # not found in GEM_HOME, check directories of GEM_PATH
+        separator = ctx.windows? ? ';' : ':'
+        paths = self.class.gem_path(ctx).split(separator)
```

Comment:
You could use `File::PATH_SEPARATOR` which is OS-aware.

Reply:
So, this could just become?
```
paths = self.class.gem_path(ctx).split(File::PATH_SEPARATOR)
```